src/Admin/Authorization.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\Module\oidc\Admin;
+
+use SimpleSAML\Error\Exception;
+use SimpleSAML\Locale\Translate;
+use SimpleSAML\Module\oidc\Bridges\SspBridge;
+use SimpleSAML\Module\oidc\Exceptions\AuthorizationException;
+use SimpleSAML\Module\oidc\Services\AuthContextService;
+
+class Authorization
+{
+    public function __construct(
+        protected readonly SspBridge $sspBridge,
+        protected readonly AuthContextService $authContextService,
+    ) {
+    }
+
+    public function isAdmin(): bool
+    {
+        return $this->sspBridge->utils()->auth()->isAdmin();
+    }
+
+    /**
+     * @throws \SimpleSAML\Module\oidc\Exceptions\AuthorizationException
+     */
+    public function requireAdmin(bool $forceAdminAuthentication = false): void
+    {
+        if ($forceAdminAuthentication) {
```

Comment:
`=== true` strict comparison.
Maybe this is less relevant with a `bool` typed param, but it's a defense-in-depth